actions/maven/publish/action.yml

```
@@ -0,0 +1,117 @@
+# Copyright 2023 SLSA Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+inputs:
+  provenance-download-name:
+    description: "The artifact name for the package provenance."
+    required: true
+    type: string
+  provenance-download-sha256:
+    description: "The sha256 of the package provenance artifact."
+    required: false
+    type: string
+  target-download-sha256:
+    description: "The sha256 of the target directory."
+    required: true
+    type: string
+  maven-username:
+    description: "Maven username"
+    required: false
+  maven-password:
+    description: "Maven password"
+    required: false
+  gpg-key-pass:
+    description: "gpg-key-pass"
+    required: false
+  gpg-private-key:
+    description: "gpg-key-pass"
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the project repository
+      uses: slsa-framework/slsa-github-generator/.github/actions/secure-project-checkout@v1.8.0-rc.1
+    - name: Set up Java for publishing to Maven Central Repository
+      uses: actions/setup-java@5ffc13f4174014e2d4d4572b3d74c3fa61aeb2c2 # v3
+      env:
+        MAVEN_USERNAME: ${{ inputs.maven-username }}
+        MAVEN_PASSWORD: ${{ inputs.maven-password }}
+        GPG_KEY_PASS: ${{ inputs.gpg-key-pass }}
+      with:
+        java-version: '11'
+        distribution: 'temurin'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+        gpg-private-key: ${{ inputs.gpg-private-key }}
+        gpg-passphrase: GPG_KEY_PASS
```

Are these supposed to be the inputs? Why not this?

```suggestion
        server-username: ${{ inputs.maven-username }}
        server-password: ${{ inputs.maven-password }}
        gpg-private-key: ${{ inputs.gpg-private-key }}
        gpg-passphrase: ${{ inputs.gpg-key-pass }}
```

---

Yes, that is intended. The parameters have to be env names which then contain the values. Otherwise we run into [this problem](https://stackoverflow.com/questions/74909075/unable-to-decrypt-gpg-passphrase-with-org-apache-maven-pluginsmaven-gpg-plugin#:~:text=Solution%20to%20your%20problem&text=gpgPassphrase%20should%20be%20the%20name,MAVEN_GPG_PASSPHRASE%20).).